### PR TITLE
Check for unused dependencies in lint pipeline

### LINKED ISSUE #1761: Add DepCheck in Core mono repo
Depcheck is a tool for analyzing the dependencies in a project to see: how each dependency is used, which dependencies are useless, and which dependencies are missing from package.json.

This tool is great for some quick check of project dependencies.

### PR DESCRIPTION
**Pull request by mcmire**
## Explanation

This matches what we do for other projects (e.g., https://github.com/MetaMask/metamask-module-template/blob/085b701d9084b45179caef9db1c0b60afa659a78/package.json#L32-L35).

## References

Fixes #1761.

## Changelog

(No consumer-facing changes)

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate

### CODE REVIEW

File: .depcheckrc.json
Comment (Gudahtt) [inline .depcheckrc.json:1]: Nit: It'd be really nice if we could add comments here to explain why each package is ignored. In extension we use `.yml` which allows comments. Maybe it's worth switching formats here and in the module template as well

**Review by Gudahtt — APPROVED** [review]
LGTM!

File: .depcheckrc.json
Comment (mcmire) [reply .depcheckrc.json:1]: I was thinking the same thing. This isn't high priority, so I can make that change in the module template.

**Review by Gudahtt — APPROVED** [review]

**Comment by mcmire** [comment]
This is ready for review again!

File: packages/eth-json-rpc-provider/package.json [hunk 1]
```diff
@@ -45,7 +45,6 @@
     "@metamask/auto-changelog": "^3.4.3",
     "@types/jest": "^27.4.1",
     "deepmerge": "^4.2.2",
-    "depcheck": "^1.4.3",
```
Comment (mcmire) [inline packages/eth-json-rpc-provider/package.json:48]: This package and `json-rpc-engine` actually have extraneous scripts and dependencies — I will make another PR to correct this.

**Pull request merged**